### PR TITLE
Issue 3341: Fix build errors and warnings in `common`

### LINKED ISSUE #3341: Clean up warnings in build for pravega
Problem description
    Modules in the pravega build have a large number of warnings. 

  Pravega/pravega modules

Make modifications to eliminate the warnings - hopefully to improve build time and clarity. In a couple of cases the code examples in Javadoc were truncated.

### PR DESCRIPTION
**Pull request by huide9**
**Change log description**  
Fix annoying build warnings/errors during the javadoc compilation for "common"

**Purpose of the change**  
fixed 18 errors and 4 warnings
fixes #3341

**What the code does**  
java 8 has more strict xlint rule, follow the rule add necessary information in java comments

**How to verify it**  
cd pravega
./gradlew common:build

** Note ** 
the changes depend on build.gradle changes in PR#3359; it's not merged at the moment so the file build.gradle is also attached.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -119,7 +119,13 @@ project('common') {
             project(':common').sourceSets.main.java
             project(':shared:protocol').sourceSets.main.java
         }
+        dependencies {
```
Comment (andreipaduroiu) [inline build.gradle]: You're adding a runtime dependency on Lombok. This has been removed with issue #3289. Please do not add them back.

File: common/src/main/java/io/pravega/common/io/serialization/VersionedSerializer.java [hunk 2]
```diff
@@ -475,7 +475,7 @@ void deserializeContents(InputStream stream, ReaderType target) throws IOExcepti
      *
      * Example:
      * <pre>
-     *
+     * {@code
```
Comment (andreipaduroiu) [inline common/src/main/java/io/pravega/common/io/serialization/VersionedSerializer.java]: @sachin-j-joshi removed this in #3116 to reduce warnings. Why are you adding it back?

Comment (elizabethbain) [reply common/src/main/java/io/pravega/common/io/serialization/VersionedSerializer.java]: The @code issue causes warnings when javadoc can't distinguish between the code annotations and its own annotations in java 8. There is a workaround I found where the code snippet can be modified like this:
<pre><code>
...... your code snippet here with
     {@literal @}your annotation here
''
</code><pre>
Verified it does what you want in javadoc. Before it was truncating the code segment. Literal tells it to take it literally and not interpret it as a javadoc annotation.

Comment (elizabethbain) [reply common/src/main/java/io/pravega/common/io/serialization/VersionedSerializer.java]: hmmm..... did that truncate my example fix?

File: build.gradle (hunk 1 above)
Comment (tkaitchuck) [reply build.gradle]: No. This is inside of the 'javadoc' block, so it should not affect the main dependencies above.

Comment (huide9) [reply build.gradle]: it's actually Tom's code, it did fixed hundreds of build errors/warnings; and based on this changes the left 18 errors get fixed. looking #3289 ...

**Comment by huide9** [comment]
The changes are covered by #3365 and I was requested by Bath to not merge the changes made for "common" here.

**Comment by huide9** [comment]
> This PR has multiple problems even before I look at the code: it doesn't refer to an issue, it is missing the sign off, and it is not properly closing one or more issues. Please make sure you have read and understood the contribution guidelines:
> 
> https://github.com/pravega/pravega/wiki/Contributing

Fixed,  please let me know for any further concerns.

File: build.gradle (hunk 1 above)
Comment (fpj) [inline build.gradle]: I don't understand why these changes in `build.gradle` are showing up here, they are not part of this PR, right?

File: common/src/main/java/io/pravega/common/lang/ProcessStarter.java [hunk 3]
```diff
@@ -48,6 +48,7 @@ private ProcessStarter(Class<?> target) {
      * Creates a new instance of the ProcessStarter class.
      *
      * @param target The Class to start. This class must have a static main() method defined in it.
+     * @return ProcessStarter
```
Comment (fpj) [inline common/src/main/java/io/pravega/common/lang/ProcessStarter.java]: Can we add something more descriptive like:

```
@return returns an instance of {@link ProcessStarter}
```

**Comment by fpj** [comment]
@huide9 @elizabethbain Is this PR superseded by PR #3365 ?

**Comment by huide9** [comment]
close as it's duplicate with #3365
